### ISSUE TITLE
what is the warning mean?

### ISSUE THREAD

**Issue by cereschen** [ASDAlexander77/TypeScriptCompiler#92]
This code has no errors, but it still reports this warning. I guess it's related to generics

`<unknown>:0: warning: types have different sizes:
 '!llvm.struct<packed (ptr<i8>, i1)>' size of #9,
 '!llvm.struct<packed (ptr<i8>, ptr<i8>)>' size of #16
<unknown>:0: warning: using casting to undefined value`

```ts
class LocalMap<T, R> {
  #storage: Array<[T, R]> = [];
  get(key: T): R | null {
    for (let i = 0; i < this.#storage.length; i++) {
      if (this.#storage[i][0] === key) {
        return this.#storage[i][1];
      }
    }
    return null
  }

  set(key: T, value: R): this {
    for (let i = 0; i < this.#storage.length; i++) {
      if (this.#storage[i][0] === key) {
        this.#storage[i][1] = value;
        return this;
      }
    }
    this.#storage.push([key, value]);
    return this;
  }
  forEach(callbackFn: (value: R, key: T) => void) {
    for (let i = 0; i < this.#storage.length; i++) {
      callbackFn(this.#storage[i][1], this.#storage[i][0]);
    }
  }
  *[Symbol.iterator]() {
    for (let i = 0; i < this.#storage.length; i++) {
      yield this.#storage[i]
    }
  }

  get size() {
    return this.#storage.length
  }
  clear() {
    this.#storage = []
  }

  delete(key: T): this {
    let index = -1
    for (let i = 0; i < this.#storage.length; i++) {
      if (this.#storage[i][0] === key) {
        index = i
        break
      }
    }
    if (index > -1) {
      let newArr: Array<[T, R]> = []
      for (let i = 0; i < this.#storage.length; i++) {
        if (i !== index) {
          newArr.push(this.#storage[i])
        }
      }
      this.#storage = newArr
    }
    return this
  }
}

void function main() {
  let obj = new LocalMap<string, boolean>()
  obj.set('1111', true)
  obj.set('2222', false)
  obj.set('3333', false)
  obj.delete('2222')
  for (let [k, v] of obj) {
    console.log(`${k} is ${v}`);
  }
}
```

**Comment by ASDAlexander77** [comment]
this warning because of using "yield", as it requires to follow the JS specification we need to return as last value ```[false, undefined]```  thus it will result in the type '''[bool, [T, R] | undefined]''' which is tuple of union type and thus compiler warning about it. You can ignore them. I have not decided myself what to do with them yet :)
